ams/ams-server/src/main/java/com/netease/arctic/ams/server/optimize/TableOptimizeItem.java

```diff
@@ -673,10 +673,14 @@ private void optimizeTasksClear(BasicOptimizeCommit optimizeCommit) {
           getMapper(sqlSession, TableOptimizeRuntimeMapper.class);
 
       try {
+        // reset snapshot id in table runtime, because the tasks were cleared rather than committed.
+        // So need to plan again.
+        tableOptimizeRuntime.setCurrentSnapshotId(TableOptimizeRuntime.INVALID_SNAPSHOT_ID);
+        tableOptimizeRuntime.setCurrentChangeSnapshotId(TableOptimizeRuntime.INVALID_SNAPSHOT_ID);
         // persist partition optimize time
         tableOptimizeRuntimeMapper.updateTableOptimizeRuntime(tableOptimizeRuntime);
       } catch (Throwable t) {
-        LOG.warn("failed to persist tableOptimizeRuntime after commit, ignore. " + getTableIdentifier(), t);
+        LOG.warn("failed to persist tableOptimizeRuntime after commit failed, ignore. " + getTableIdentifier(), t);
```

Comment:
There is a standard procedure to update tableOptimizeRuntime to keep the consistency of memory and sysdb.

We should stick to it.

The procedure is like
```
oldRuntime = tableOptimizeRuntime.clone();
try {
    // update tableOptimizeRuntime

} catch () {
    tableOptimizeRuntime.restore(oldRuntime);
    sqlSession.rollback(true);
}
```

BTW, there are too `failed` in the warn log.

Reply:
OK